benchexec/tools/sv_benchmarks_util.py

```diff
@@ -40,3 +42,132 @@ def get_data_model_from_task(task, param_dict):
+
+    @param task: An instance of a task
+    @return: List of witness files
+    """
+    witness_files, _ = _partition_input_files(
+        task.input_files_or_identifier, task.options
+    )
+    return witness_files
+
+
+def get_unique_witness(task):
+    """
+    This function returns the unique witness file from the task.
+    It raises an exception if there are multiple witness files.
+    The witness file is identified by the option "witness" in the task options.
+
+    @param task: An instance of a task
+    @return: Unique witness file
+    """
+    witness_files = get_witness_input_files(task)
+    if len(witness_files) > 1:
+        raise UnsupportedFeatureException(
+            "Tool does not support tasks with more than one witness file"
+        )
+    return witness_files[0]
+
+
+def get_non_witness_input_files_or_identifier(task):
+    """
+    This function returns the non-witness input files or the identifier from the task.
+    They consist of all elements which do not match the witness file name.
+    The witness file is identified by the option "witness" in the task options.
+
+    @param task: An instance of a task
+    @return: List of non-witness files
+    """
+    _, other_files = _partition_input_files(
+        task.input_files_or_identifier, task.options
+    )
+    return other_files
+
+
+def get_non_witness_input_files(task):
+    """
+    This function returns the non-witness input files from the task.
+    They consist of all files which do not match the witness file name.
+    The witness file is identified by the option "witness" in the task options.
+
+    @param task: An instance of a task
+    @return: List of non-witness files
+    """
+    _, other_files = _partition_input_files(task.input_files, task.options)
+    return other_files
+
+
+def get_single_non_witness_input_file(task):
+    """
+    This function returns the unique non-witness file from the task.
+    It raises an exception if there are multiple non-witness files.
+    Non-witness files consist of all files which do not match the witness file name.
+    The witness file is identified by the option "witness" in the task options.
+
+    @param task: An instance of a task
+    @return: Unique non-witness file
+    """
+    other_files = get_non_witness_input_files(task)
+    if len(other_files) > 1:
+        raise UnsupportedFeatureException(
+            "Tool does not support tasks with more than one input file"
+        )
+    return other_files[0]
+
+
+def get_witness_options(options, task, witness_options):
+    """
+    This function returns the additional options to handle witnesses.
+    It checks if the witness is passed as an option or through the task definition.
+    If the witness is passed through both, it raises an exception.
+    If no witness is given in the task, it returns an empty list.
+    Therefore, this function can be used with any task regardless if it
+    has a witness option or not.
```

Comment:
I think we should add some more high-level / general documentation. Right now I do not think that a future developer of a tool-info module would know how to implement support for the witness option in task definitions except by copying it from an existing tool-info module.

So we should write something like "To handle the witness keyword in task.options, ... append the result of get_witness_option to the options passed to the tool ... use get_..._non_witness_input_files to retrieve the list of remaining input files for your tool ..."

Maybe the best place to add is here, at least I do not know a better place.

Reply:
Now this should be part of `handle_witness_of_task`, of course. And it can be shorter because everything is now much easier to understand from the point of view of a caller.